llvm/test/CodeGen/X86/sm4-evex-intrinsics.ll

```diff
@@ -0,0 +1,64 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: llc < %s -verify-machineinstrs -mtriple=x86_64-- --show-mc-encoding -mattr=+sm4,+avx10.2-512 | FileCheck %s
+; RUN: llc < %s -verify-machineinstrs -mtriple=i686-- --show-mc-encoding -mattr=+sm4,+avx10.2-512 | FileCheck %s
+
+define <4 x i32> @test_int_x86_vsm4key4128(<4 x i32> %A, <4 x i32> %B) {
+; CHECK-LABEL: test_int_x86_vsm4key4128:
+; CHECK:       # %bb.0:
+; CHECK-NEXT:    vsm4key4 %xmm1, %xmm0, %xmm0 # encoding: [0xc4,0xe2,0x7a,0xda,0xc1]
```

Comment:
I assume there's a comment about EVEX2VEX compress.

Reply:
All machines supports SM4 EVEX version also supports SM4 VEX version. similar situations for AVX512F.

Reply:
So there will be a EVEX2VEX comment. The point is we need to select to EVEX instruction first and then optimize to VEX.

Reply:
[39513cd](https://github.com/llvm/llvm-project/pull/113402/commits/39513cdd193f3dc48e7df635b50c70ea51f2adef)